src/windows-emulator/windows_emulator.hpp

```diff
@@ -17,10 +17,13 @@ struct io_device;
 
 #define opt_func utils::optional_function
 
-struct emulator_callbacks : module_manager::callbacks, process_context::callbacks
+struct emulator_callbacks : process_context::callbacks
 {
     using continuation = instruction_hook_continuation;
 
+    std::vector<std::function<void(mapped_module& mod)>> on_module_load{};
+    std::vector<std::function<void(mapped_module& mod)>> on_module_unload{};
```

Comment:
just use an opt_func like for the other callbacks. using a vector is unnecessary. it just costs performance overhead while likely being unneeded. usually, one callback is enough, and if not, it can be implemented ontop by the consumer